ipcore/avalon_wrapper.py

```diff
@@ -377,10 +383,10 @@ def channel_to_sample(avalon_sink_data_signal, dataplane_sink_data_tmp_signal, d
     return f"""
     if rising_edge(clk) then
         if avalon_sink_valid = '1' then
-            if avalon_sink_channel = "00" then
+            if avalon_sink_channel = "0" then
```

Comment:
It looks like the generated HDL still hardcodes some assumptions about the number of channels and the width of the channel data type.

It seems like we could use a loop over the number of channels to generate the code more generically. Using a case statement instead of if-elsif should help make the generation more generic as well. 

We should be able to use our `num_to_bitstring` routine to handle not hardcoding the width of the channel literals. 
```